core/modes/language_modes.py

```diff
@@ -74,20 +20,162 @@
 mod.tag("code_language_forced", "This tag is active when a language mode is forced")
 mod.list("language_mode", desc="Name of a programming language mode.")
 
-ctx.lists["self.language_mode"] = {
-    name: language
-    for language in language_extensions
-    for name in language_name_overrides.get(language, [language])
-}
-
 # Maps extension to languages.
-extension_lang_map = {
-    "." + ext: language
-    for language, extensions in language_extensions.items()
-    for ext in extensions.split()
-}
-
-language_ids = set(language_extensions.keys())
+extension_lang_map = None
+
+language_ids = None
+language_extensions = None
+
+SETTINGS_FILENAME = "language_modes.csv"
+settings_filepath = compute_csv_path(SETTINGS_FILENAME)
+
+LANGUAGE_HEADERS = ["language", "extensions", "spoken_forms"]
+
+
+def make_sure_settings_file_exists():
+    # Maps language mode names to the extensions that activate them and language spoken forms. Only put things
+    # here which have a supported language mode; that's why there are so many
+    # commented out entries. TODO: make this a csv file?
```

Comment:
Please update these comments to reflect the current state of things.

Reply:
I tried to update the comments to reflect the current state of things. Let me know if you think I missed anything.